pkg/component/kubeproxy/resources.go

```diff
@@ -654,3 +566,128 @@ func (k *kubeProxy) getMode() kubeproxyconfigv1alpha1.ProxyMode {
 	}
 	return "iptables"
 }
+
+func (k *kubeProxy) getInitContainers(kubernetesVersion *semver.Version, image string) []corev1.Container {
+	initContainers := []corev1.Container{
+		{
+			Name:            "cleanup",
+			Image:           image,
+			ImagePullPolicy: corev1.PullIfNotPresent,
+			Command: []string{
+				"sh",
+				"-c",
+				fmt.Sprintf("%s/%s %s", volumeMountPathCleanupScript, dataKeyCleanupScript, volumeMountPathMode),
+			},
+			Env: []corev1.EnvVar{
+				{
+					Name:  "KUBE_PROXY_MODE",
+					Value: string(k.getMode()),
+				},
+				{
+					Name:  "EXECUTE_WORKAROUND_FOR_K8S_ISSUE_109286",
+					Value: strconv.FormatBool(versionutils.ConstraintK8sLess125.Check(kubernetesVersion)),
+				},
+			},
+			SecurityContext: &corev1.SecurityContext{
+				Privileged: pointer.Bool(true),
+			},
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      volumeNameCleanupScript,
+					MountPath: volumeMountPathCleanupScript,
+				},
+				{
+					Name:      volumeNameKernelModules,
+					MountPath: volumeMountPathKernelModules,
+				},
+				{
+					Name:      volumeNameDir,
+					MountPath: volumeMountPathDir,
+				},
+				{
+					Name:      volumeNameMode,
+					MountPath: volumeMountPathMode,
+				},
+				{
+					Name:      volumeNameKubeconfig,
+					MountPath: volumeMountPathKubeconfig,
+				},
+				{
+					Name:      volumeNameConfig,
+					MountPath: volumeMountPathConfig,
+				},
+			},
+		},
+	}
+
+	k8sGreaterEqual129 := versionutils.ConstraintK8sGreaterEqual129.Check(kubernetesVersion)
+	if k8sGreaterEqual129 {
+		initContainers = append(initContainers, k.getKubeProxyContainer(k8sGreaterEqual129, image, true))
+	}
+
+	return initContainers
+}
+
+func (k *kubeProxy) getKubeProxyContainer(k8sGreaterEqual129 bool, image string, init bool) corev1.Container {
+	container := corev1.Container{
+		Name:            containerName,
```

Comment:
I don't know how you tested this but:
```yaml
  - lastTransitionTime: "2024-01-11T08:21:58Z"
    lastUpdateTime: "2024-01-11T08:21:58Z"
    message: 'error during apply of object "apps/v1/DaemonSet/kube-system/kube-proxy-local-v1.29.0":
      DaemonSet.apps "kube-proxy-local-v1.29.0" is invalid: spec.template.spec.initContainers[1].name:
      Duplicate value: "kube-proxy"'
    reason: ApplyFailed
    status: Progressing
    type: SystemComponentsHealthy
```

It seems that you cannot have container and initContainer with the same name.